lib/httparty/request.rb

```diff
@@ -276,11 +276,10 @@ def get_charset
     end
 
     def encode_with_ruby_encoding(body, charset)
-      if !body.nil? && Encoding.name_list.include?(charset)
-        body.force_encoding(charset)
-      else
-        body
-      end
+      encoding = Encoding.find(charset)
+      body.force_encoding(charset)
+    rescue ArgumentError
```

Comment:
Actually, it looks like this method can never get called with a `nil` body: https://github.com/jnunemaker/httparty/blob/master/lib/httparty/request.rb#L350

Do you still want an explicit `nil?` check in this method?

I can add a test anyway—it seems reasonable to have a “response body is nil/empty” test—but that test would never actually exercise this method.